dowhy/causal_estimator.py

```diff
@@ -515,24 +559,13 @@ def _test_significance_with_bootstrap(self, data: pd.DataFrame, estimate_value,
                 null_estimates, {"num_null_simulations": num_null_simulations, "sample_size_fraction": 1}
             )
 
-        # Processing the null hypothesis estimates
-        sorted_null_estimates = np.sort(self._bootstrap_null_estimates.estimates)
-        self.logger.debug("Null estimates: {0}".format(sorted_null_estimates))
-        median_estimate = sorted_null_estimates[int(num_null_simulations / 2)]
-        # Doing a two-sided test
-        if estimate_value > median_estimate:
-            # Being conservative with the p-value reported
-            estimate_index = np.searchsorted(sorted_null_estimates, estimate_value, side="left")
-            p_value = 1 - (estimate_index / num_null_simulations)
-        if estimate_value <= median_estimate:
-            # Being conservative with the p-value reported
-            estimate_index = np.searchsorted(sorted_null_estimates, estimate_value, side="right")
-            p_value = estimate_index / num_null_simulations
-        # If the estimate_index is 0, it depends on the number of simulations
-        if p_value == 0:
-            p_value = (0, 1 / len(sorted_null_estimates))  # a tuple determining the range.
-        elif p_value == 1:
-            p_value = (1 - 1 / len(sorted_null_estimates), 1)
+        # This calculates a two-sided percentile p-value
+        # See footnotes in https://journals.sagepub.com/doi/full/10.1177/2515245920911881
+        half_p_value = np.mean(
```

Comment:
We shouldn't need to repeat the same code here? Can we reuse or move out the functionality?